Replace deprecated ProductLoaderCriteriaEvent

JuicyLung91 opened the pull request:
Add ProductPageCriteriaEvent class for Subscriber because of deprecated ProductLoaderCriteriaEvent

<!--
Thank you for contributing to Shopware! Please fill out this description template to help us to process your pull request.

Please make sure to fulfil our contribution guideline (https://docs.shopware.com/en/shopware-platform-dev-en/contribution/contribution-guideline?category=shopware-platform-dev-en/contribution).

Do your changes need to be mentioned in the documentation?
Add notes on your change right now in the documentation files in /src/Docs/Resources and add them to the pull request as well. 
-->

### 1. Why is this change necessary?
Because the ProductLoaderCriteriaEvent Event is deprecated since v6.4.0

### 2. What does this change do, exactly?
Replaces the deprectaed fiel with a new one.

shyim commented:
Has been fixed with https://github.com/shopware/shopware/commit/05cc2bd8f27fc2f0c39b377e64753f7e26e6e082#diff-f67c2b139e7f0abbd2bf29a57997b459R18

JuicyLung91 committed:
Replace deprecated ProductLoaderCriteriaEvent

Add ProductPageCriteriaEvent class for Subscriber because of deprecated ProductLoaderCriteriaEvent

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/shopware/platform?pullRequest=1577) <br/>All committers have signed the CLA.

mitelg commented on src/Docs/Resources/current/50-how-to/010-indepth-guide-bundle/080-storefront.md:
```diff
@@ -27,15 +27,15 @@ Just use the constant mentioned above and choose a method name to call once the
+            ProductPageCriteriaEvent::class => 'onProductCriteriaLoaded'
         ];
     }
 
```
there are three more occurrences which needs to be fixed

L42
L48
L52

JuicyLung91 committed:
Add ProductPageCriteriaEvent on missed lines

J-Rahe commented:
Hey @JuicyLung91, thanks for your Contribution!
